src/storage.js

```diff
@@ -58,15 +58,27 @@ class Storage {
         // The first event we get will be sent immediatelly,
         // other will be sent when MAX_EVENTS is reached or when we reach the end of an interval.
         this._atFirstEvent = true;
+
+        const collectEvents = () => {
+            this._collectEvents(this._session.id())
+        };
         // Set up an interval to send evenst periodically
         // TODO: Make sure using setInterval is not a terrible idea
-        const collectOnInterval = () => this._collectEvents(this._session.id());
-        this._interval = setInterval(collectOnInterval, EVENTS_PING_INTERVAL);
+        this._interval = setInterval(collectEvents, EVENTS_PING_INTERVAL);
 
         // If persisted events have reached limit, submit them
         if (this._events.length >= MAX_EVENTS) {
-            this._collectEvents();
+            collectEvents()
         }
+
+        // If the page unloads we want to collect any events
+        // in case the user never comes back to this page.
+        //
+        // TODO: even though this *usually* (focus on the usually, sometimes is doesn't)
```

Comment:
Mh, that's odd that sometimes it doesn't. Network errors? Because the keepalive should do that

Reply:
My hunch is that since collectEvents does some stuff before the actual uploading maybe the page is closed before that? Since this `unload` is probably a callback at the end of `beforeunload`, this shouldn't be a problem, but who know...

Reply:
Yeah, I see.